core/core.php

```diff
@@ -181,4 +185,106 @@ public function validate_badge_order($user_id)
 
 		return true;
 	}
+
+	/**
+	 * Gets the url to the user avatar.
+	 * Returns false on failure or if there is no avatar.
+	 *
+	 * @param array $user_data Avatar data from the user table
+	 * @return array|boolean
+	 */
+	public function get_avatar_data($user_data)
+	{
+		if (!$this->config['allow_avatar'])
+		{
+			return false;
+		}
+
+		$row = [
+			'avatar' => $user_data['user_avatar'],
+			'avatar_width' => $user_data['user_avatar_width'],
+			'avatar_height' => $user_data['user_avatar_height'],
+		];
+
+		$avatar_data = [
+			'src' => $row['avatar'],
+			'width' => $row['avatar_width'],
+			'height' => $row['avatar_height'],
+		];
+
+		$driver = $this->avatar_manager->get_driver($user_data['user_avatar_type']);
+
+		if (!$driver)
+		{
+			return false;
+		}
+
+		$avatar_data = $driver->get_data($row);
+
+		if ($user_data['user_avatar_type'] === 'avatar.driver.gravatar')
+		{
+			$avatar_data['src'] = $this->get_gravatar_url($row);
+		} else if ($user_data['user_avatar_type'] === 'avatar.driver.upload') {
```

Comment:
Convention within this extension (and probably phpbb more generally?) appears to be to always have braces in their own line, so can we please expand these to
```php
}
else if (...)
{
```
?